state/factory/factory_test.go

```diff
@@ -303,6 +303,22 @@ func TestState(t *testing.T) {
 	testState(sf, t)
 }
 
+func TestHistoryState(t *testing.T) {
+	testTrieFile, _ := ioutil.TempFile(os.TempDir(), triePath)
+	cfg := config.Default
+	cfg.Chain.TrieDBPath = testTrieFile.Name()
+	cfg.Chain.EnableHistoryStateDB = true
```

Comment:
add a unit test: if this is false, we should expect an error, say ErrNoArchiveData